### PR TITLE
Added: BookController, BookDto layer, BookMapper, findById method to …

### PR DESCRIPTION
**Pull request by pavlov-n**
…bookRepository and bookService. Updated: pom.xml file(added dependencies for mapper).

### CODE REVIEW

File: src/main/java/com/bookshop/config/MapperConfig.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+import org.mapstruct.InjectionStrategy;
+import org.mapstruct.NullValueCheckStrategy;
+
+@org.mapstruct.MapperConfig(
+        componentModel = "spring",
+        injectionStrategy = InjectionStrategy.CONSTRUCTOR,
+        nullValueCheckStrategy = NullValueCheckStrategy.ALWAYS,
+        implementationPackage = "<PACKAGE_NAME>.impl"
+)
```
Comment (liliia-ponomarenko) [inline src/main/java/com/bookshop/config/MapperConfig.java]: it is better to specify @MapperConfig directly above the mapper class

File: src/main/java/com/bookshop/dto/BookDto.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
+package com.bookshop.dto;
+
+import java.math.BigDecimal;
+import lombok.Data;
+
+@Data
+public class BookDto {
```
Comment (liliia-ponomarenko) [inline src/main/java/com/bookshop/dto/BookDto.java]: better use record

File: src/main/java/com/bookshop/dto/CreateBookRequestDto.java [hunk 3]
```diff
@@ -0,0 +1,14 @@
+package com.bookshop.dto;
+
+import java.math.BigDecimal;
+import lombok.Data;
+
+@Data
+public class CreateBookRequestDto {
```
Comment (liliia-ponomarenko) [inline src/main/java/com/bookshop/dto/CreateBookRequestDto.java]: the same

File: src/main/java/com/bookshop/service/impl/BookServiceImpl.java [hunk 4]
```diff
@@ -1,26 +1,41 @@
...
+                .toList();
+    }
+
+    @Override
+    public BookDto findById(Long id) {
+        Book book = bookRepository.findById(id)
+                .orElseThrow(()
+                        -> new EntityNotFoundException(
+                                "can't find book with such id: %d".formatted(id)));
+        return bookMapper.toDto(book);
```
Comment (liliia-ponomarenko) [inline src/main/java/com/bookshop/service/impl/BookServiceImpl.java]: we can make use of Optional's map method here:

```
return repository.findById(id)
            .map(bookMapper::toDto)
            .orElseThrow(...);
```

**Review by liliia-ponomarenko — CHANGES_REQUESTED** [review]
Good job! Let’s  improve your solution ;)

File: src/main/java/com/bookshop/mapper/BookMapper.java [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+import org.mapstruct.NullValueCheckStrategy;
+
+@Mapper(componentModel = "spring",
+        injectionStrategy = InjectionStrategy.CONSTRUCTOR,
+        nullValueCheckStrategy = NullValueCheckStrategy.ALWAYS,
+        implementationPackage = "<PACKAGE_NAME>.impl")
+public interface BookMapper {
+    BookDto toDto(Book book);
+
+    Book toModel(CreateBookRequestDto requestDto);
```
Comment (NazarSmith228) [inline src/main/java/com/bookshop/mapper/BookMapper.java]: `toBook` sounds better :)

File: src/main/java/com/bookshop/repository/impl/BookRepositoryImpl.java [hunk 6]
```diff
@@ -1,38 +1,44 @@
...
+        try (Session session = sessionFactory.openSession()) {
+            return session.createQuery("FROM Book", Book.class).getResultList();
+        } catch (Exception e) {
+            throw new EntityNotFoundException("Can't get all books from DB", e);
+        }
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        return sessionFactory.fromSession(s -> Optional.ofNullable(s.get(Book.class, id)));
```
Comment (NazarSmith228) [inline src/main/java/com/bookshop/repository/impl/BookRepositoryImpl.java]: you might also need to handle possible exception here

**Review by NazarSmith228 — APPROVED** [review]
LGTM, but please pay attention to my remarks

**Pull request merged**